### PR TITLE
Fb asim etk pyqt5 long lived prusse 2021 03 10 base

### PR DESCRIPTION
**Pull request by prusse-martin**
- Facilitates serialization of "table contents input";
- Allow external code to be notified by cell bounds changes;
- Allow to optionally define the "id" of a cell when creating;

### CODE REVIEW

File: examples/drag_and_drop/main.py [hunk 1]
```diff
@@ -121,13 +121,18 @@ def on_terminal_changed_handler(
         def on_cells_removed_handler(cell_ids):
             print(f'removed {cell_ids}')
 
+        def on_cells_bounds_changed_handler(changed_cell_bounds):
+            print(f'cells bounds changed {changed_cell_bounds}')
```
Comment (williamjamir) [inline examples/drag_and_drop/main.py:125]: @prusse-martin

Comment (prusse-martin) [reply examples/drag_and_drop/main.py:125]: `examples/drag_and_drop/main.py `

File: qmxgraph/cell_bounds.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+"""
+Classes to used to represent bounds in:
```
Comment (oliveira-mauricio) [inline qmxgraph/cell_bounds.py:2]:
```suggestion
Classes used to represent bounds in:
```

File: qmxgraph/cell_bounds.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+
+@attr.s(frozen=True, slots=True)
+class ParentAnchorPosition:
+    """
+    This represents the anchor point relative to the parent cell.
+    See :class:`CellBounds` for more details.
+
+    The attributes are normalized to the parent bounds. For example, give
+    a parent with positioned at the x coordinate `50` with width `16` we
```
Comment (oliveira-mauricio) [inline qmxgraph/cell_bounds.py:28]:
```suggestion
    a parent positioned at the x coordinate `50` with width `16` we
```

File: qmxgraph/page/graphs.js [hunk 4]
```diff
@@ -64,6 +64,36 @@ graphs.createGraph = function createGraph (container, options, styles) {
...
+        if (
+            (userObject !== null)
+            && (userObject.nodeType === mxConstants.NODETYPE_ELEMENT)
+            && userObject.hasAttribute(name)
+        ) {
+            userObject.removeAttribute(name);
+        } else {
+            throw Error(
+                "The cell " + this.getId()
+                + " do not have an attribute named " + name
```
Comment (oliveira-mauricio) [inline qmxgraph/page/graphs.js:79]:
```suggestion
                + " does not have an attribute named " + name
```

**Review by oliveira-mauricio — APPROVED** [review]
Just noticed a few docs to fix.

**Pull request merged**